examples/30_extended/fetch_runtimes_tutorial.py

```diff
@@ -325,3 +357,73 @@ def get_incumbent_trace(trace):
+#
+# Running a Decision Tree model that doesn't support parallelism implicitly, but
+# using OpenML to parallelize evaluations for the outer-cross validation folds.
+
+dt = DecisionTreeClassifier()
+run6 = openml.runs.run_model_on_task(
+    model=dt, task=task, upload_flow=False, avoid_duplicate_runs=False, n_jobs=2
+)
+measures = run6.fold_evaluations
+print_compare_runtimes(measures)
+
+################################################################################
+# Running a Neural Network from scikit-learn that uses scikit-learn independent
+# parallelism using libraries such as MKL, OpenBLAS or BLIS.
+
+mlp = MLPClassifier(max_iter=10)
+run7 = openml.runs.run_model_on_task(
+    model=mlp, task=task, upload_flow=False, avoid_duplicate_runs=False
+)
+measures = run7.fold_evaluations
+print_compare_runtimes(measures)
+
+################################################################################
+# Summmary
+# *********
+# OpenML records model runtimes for the CPU-clock and the wall-clock times. The above
+# examples illustrated how these recorded runtimes can be extracted, especially when
+# using a scikit-learn model and under parallel setups too. To summarize, OpenML measures the:
+#
+# * `CPU-time` & `wallclock-time` for the whole run
+#
+#   * A run here corresponds to a call to `run_model_on_task` or `run_flow_on_task`
+#   * The recorded time is for the model fit for each of the outer-cross validations folds,
+#     i.e., the OpenML data splits
+#
+# * Python's `time` module is used to compute the runtimes
+#
+#   * `CPU-time` is recorded using the responses of `time.process_time()`
+#   * `wallclock-time` is recorded using the responses of `time.time()`
+#
+# * The timings recorded by OpenML per outer-cross validation fold is agnostic to
+#   model parallelisation
+#
+#   * The wallclock times reported in Case 2 above highlights the speed-up on using `n_jobs=-1`
+#     in comparison to `n_jobs=2`, since the timing recorded by OpenML is for the entire
+#     `fit()` procedure, whereas the parallelisation is performed inside `fit()` by scikit-learn
+#   * The CPU-time for models that are run in parallel can be difficult to interpret
+#
+# * `CPU-time` & `wallclock-time` for each search per outer fold in an HPO run
+#
+#   * Reports the total time for performing search on each of the OpenML data split, subsuming
+#     any sort of parallelism that happened as part of the HPO estimator or the underlying
+#     base estimator
+#   * Also allows extraction of the `refit_time` that scikit-learn measures using `time.time()`
+#     for retraining the model per outer fold, for the best found configuration
+#
+# * `CPU-time` & `wallclock-time` for models that scikit-learn doesn't parallelize
+#
+#   * Models like Decision Trees don't parallelize and thus both the wallclock and CPU times are
+#     accurate runtime representations as long as `n_job=1` or `n_jobs=None` at the OpenML call
```

Comment:
Are you sure? Shouldn't it be accurate even when `n_job>1` as the time is measured inside each fold independently?

Reply:
Yes, that's true. I'll remove the mention of `n_jobs`.

Reply:
I agree, shouldn't make any difference for Decision Trees and have similar CPU and wallclock times.